File: src/main/scala/com/github/mumoshu/ssp/scala/fruit/Fruit.scala
package com.github.mumoshu.ssp.scala.fruit

/**
 * val は「一定の値」で、defは「呼び出し毎に結果が変わるかもしれない関数」
 */
abstract class Fruit {
  val v: String
  def m: String
}

File: src/main/scala/com/github/mumoshu/ssp/java/fruit/Fruit.java
package com.github.mumoshu.ssp.java.fruit;

public abstract class Fruit {

    // val v: String
    // Javaにはvalとdefに相当する区別が不可能
    public abstract String v();

    // def m: String
    // Javaでは普通のメソッド定義
    public abstract String m();
}
